drivers/serial/uart_emul.c

```diff
@@ -31,9 +38,16 @@ struct uart_emul_data {
 
 	struct ring_buf *tx_rb;
 	struct k_spinlock tx_lock;
+
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	bool irq_en;
```

Comment:
Considering that there is a change on the horizon introducing interrupt driven transmit maybe rename to `rx_irq_en`?

Reply:
Fixed!